### PR TITLE
Add SSI requirements.json file

### PR DESCRIPTION
**Pull request by iamluc**
### Description

~~**Draft, as the RFC might still change.**~~

Implementation of the [RFC](https://docs.google.com/document/d/1TezwaDF_-TXQFHfVTySpd7FvSPRwSTQmSBG2ot129PE/edit?pli=1)

~~Tested manually on `allow_tests.json`/`block_tests.json` using the validator introduced [in this PR](https://github.com/DataDog/auto_inject/pull/380).~~

~~This PR does not add a CI job to check the file [as one will be added to the shared-pipeline](https://github.com/DataDog/dd-trace-dotnet/pull/5928/files#r1738550008)~~

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: loader/packaging/requirements.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            "x64": {
+                "supported": true,
+                "glibc": {
+                    "supported": true,
+                    "description": "From centOS 7",
+                    "min": "2.17.0"
+                },
+                "musl": {
+                    "supported": true,
+                    "description": "From alpine 3.19"
```
Comment (andrewlock) [inline loader/packaging/requirements.json:14]: In which case, should it be this?
```suggestion
                    "description": "From alpine 3.19",
                    "min": "1.2.4"
```

**Review by andrewlock — APPROVED** [review]
Nice 💪

File: loader/packaging/requirements.json (hunk 1 above)
Comment (iamluc) [reply loader/packaging/requirements.json:14]: Currently, the injector does not detect the musl version, so enforcing a minimum version results in a failure

Comment (andrewlock) [reply loader/packaging/requirements.json:14]: Well, that's problematic :eyetwitch:

**Review by bwoebi — APPROVED** [review]
Looks good :-)

**Pull request merged**